docker/Dockerfile.prod

```diff
@@ -1,19 +1,42 @@
-FROM node:16.14.0 as base
+# First, install all dependencies, including devDependencies, to run the build process
+FROM node:16.14.0 as build
 
-ENV NPM_CONFIG_PREFIX=/home/node/.npm-global
+WORKDIR /parabol
 
-RUN mkdir -p /parabol/node_modules && mkdir -p /home/node/.npm-global
+# Only copy dependency-related files here (vs. COPY . .) to avoid breaking the cache and running
+# the slow `yarn install` more than necessary
```

Comment:
This might seem overly verbose, but having the docker layer caching here is beneficial. By using this pattern, the only time `yarn install` needs to run is:

- if the layer is not cached (e.g., you've never built before)
- if any of the `package.json` files or `yarn.lock` changes

Otherwise, docker knows `yarn install` will produce the exact same resulting layer so it'll use the cache.